rust/src/message/sum.rs

```diff
@@ -24,14 +21,21 @@ impl SumMessageBuffer<Vec<u8>> {
             bytes: vec![0_u8; len],
         }
     }
+}
 
-    /// Create a sum message buffer from `bytes`. Fails if the `bytes` don't conform to the expected
-    /// sum message length `exp_len`.
-    fn try_from(bytes: Vec<u8>, exp_len: usize) -> Result<Self, PetError> {
-        if bytes.len() != exp_len {
-            return Err(PetError::InvalidMessage);
+impl TryFrom<Vec<u8>> for SumMessageBuffer<Vec<u8>> {
```

Comment:
Couldn't this be made generic? `T: AsRef[u8]`?

Reply:
good idea, i'll try that.

Reply:
for the records, since that doesn't work due to potential impl conflicts, these are the links @little-dude found regarding this topic:
- https://github.com/rust-lang/rust/issues/50133
- https://old.reddit.com/r/rust/comments/c1yubv/why_this_code_does_not_compile/ergj6hw/